src/guides/resolve-zil-domain.md

```diff
@@ -0,0 +1,203 @@
+---
+description: >-
+  This tutorial will teach you how to make a JSON RPC call to our Zilliqa smart
+  contract to get various records as well as owner address of a domain owner.
+---
+
+# Resolving .zil domain
+
+## Steps to resolve unstoppable domain
+
+In order to fetch any data from the Zilliqa blockchain you will need to do the following:
+
+1. Get a namehash of a domain
+2. Get resolver and owner addresses
+3. Get the contract sub-state related to your domain
+
+### Namehashing
+
+Namehashing is an algorithm that converts a domain name in a classical format \(like www.example.crypto\) to ERC-721 token id. This process is described  in more details over  [here](https://docs.unstoppabledomains.com/domain-registry-essentials/namehashing)
+
+```typescript
+import hash from 'hash.js';
+
+function namehash(domain: string): string {
+    const parent =
+      '0000000000000000000000000000000000000000000000000000000000000000';
+    return '0x' + [parent]
+      .concat(
+        domain
+          .split('.')
+          .reverse()
+          .filter(label => label),
+      )
+      .reduce((parent, label) =>
+        childhash(parent, label),
+      );
+  }
+
+function childhash( parent: string, label: string): string {
+    parent = parent.replace(/^0x/, '');
+    return sha256(parent + sha256(label), 'hex',);
+  }
+
+function sha256(message: string, inputEnc?: 'hex') {
+    return hash.sha256()
+      .update(message, inputEnc)
+      .digest('hex');
+  }
+```
+
+In more simplistic human words we are hashing all of the labels of the domain with the accumulated hash of its parents in order to get a unique ERC-721 token id. 
```

Comment:
I don't think I like the idea of mentioning ERC-721 in an article about Zilliqa.

Reply:
Agree. Namehash is not ERC-721 token id by itself. It's valid for .crypto implementation only - it's not a part of standard.